### PR TITLE
Ruby on Rails Capstone 

### PR DESCRIPTION
**Pull request by DiegoMero**
This pull request is to add all the requirements and features to my project like so: 

Firs of all here is the video: https://www.loom.com/share/3da72868e3cf426eb15ee17e32da6fa2

#### Design
- You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
  - Colors.
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.

> NOTE: In these design guidelines there are several UIs that you won't need for this exercise; also, some pages are not given a design and you will create them following the design guidelines of the other pages.
Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

#### Technical requirements

- You should use Postgres as your database.
- You should use devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.
- Your database schema should reflect the following structure:

> NOTE: do not change column names visible in the diagram. You need to change the "Entity"  name according to the theme you have chosen for your project (please note: "Transaction" is a name already used by ActiveRecord, so using it as a name for your model and table will result in an error).
<p align="center">
  ERD diagram
</p>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,143 @@
...
+
+I would like to thank all my coding partners that help me through the obstacles.
+
+- Original Design by: [Gregoire Vella](https://www.behance.net/gregoirevella)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](https://github.com/DiegoMero/take-my-money-app/blob/dev/MIT.md) licensed.
```
Comment (t-yanick) [inline README.md]: - [ ] Great job working on this capstone project. A click on the MIT link redirects to an error 404 page. As you know broken links are not professional to be in our codebse. Kindly fix this issue.
Screenshot from 2023-02-18 19-44-41

File: README.md [hunk 2]
```diff
@@ -1,24 +1,143 @@
...
 
-* Services (job queues, cache servers, search engines, etc.)
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-* Deployment instructions
+## 🚀 Live Demo <a name="live-demo"></a>
 
-* ...
+> Add a link to your deployed project.
+- [Live Demo Link](https://budget-app-service.onrender.com/)
```
Comment (t-yanick) [inline README.md]: - [ ] Your live demo link does not work. Kindly check and put a link that works. When I click on the link I get a request timeout response.
image

File: db/migrate/20230217032241_create_investments.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateInvestments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :investments do |t|
+      t.string :name
+      t.float :amount
+      t.references :user, null: false, foreign_key: true
```
Comment (t-yanick) [inline db/migrate/20230217032241_create_investments.rb]: - [ ] I think the reference here should AuthorId and it should be an alias to the User table. If you remember the Blog App project you did where you had AuthorId referencing through the User as reference, I think you need to apply same knowedge here.

File: spec/requests/home_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      get '/'
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the correct template' do
+      get '/'
+      expect(response).to render_template(:index)
+    end
+  end
+end
```
Comment (t-yanick) [inline spec/requests/home_spec.rb:15]: - [ ] Kindly add units tests for models to test associations and validations

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️

Hi @DiegoMero , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Excellent video presentation
:heavy_check_mark: Use of partials
:heavy_check_mark: All pages present and functionalities as expected
:heavy_check_mark: Good documentation
:heavy_check_mark: Login and registration implemented with devise

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Using a more descriptive name for your branch. `feature` is not quite descriptive

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 5]
```diff
@@ -1,24 +1,143 @@
...
-* Services (job queues, cache servers, search engines, etc.)
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-* Deployment instructions
+## 🚀 Live Demo <a name="live-demo"></a>
 
-* ...
+> Add a link to your deployed project.
+- [Live Demo Link](https://take-my-money-app.onrender.com/)
+
```
Comment (omar-muhamad) [inline README.md]: - [ ] The same problem still exists, Please provide a valid and functioning deployment link.

image

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @DiegoMero 

Nice job making the required changes. Unfortunately, you missed some of them but you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by EricMbouwe — APPROVED** [review]
Hi @DiegoMero, 

In my opinion, your project is ready for presentation! ⚡🔥 There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**